pkg/v1/remote/descriptor.go

```diff
@@ -87,6 +88,30 @@ func Get(ref name.Reference, options ...Option) (*Descriptor, error) {
 	return get(ref, acceptable, options...)
 }
 
+// Head returns a v1.Descriptor for the given reference. This is useful for
+// getting the current digest for a reference without impacting the rate limit.
+func Head(ref name.Reference, options ...Option) (*v1.Descriptor, error) {
```

Comment:
I wouldn't mention the rate limit, since it's a registry-specific thing, just:

```
// Head returns a v1.Descriptor for the given reference by issuing a HEAD request.
```

Mentioning the HEAD request should be enough of a clue that this works around the docker hub rate limit.

I would update the comment for `Get` to mention that the response includes the manifest body,  and something like: "See Head if you don't need the response body."